### PR TITLE
[Magiclysm] Tune enhancement spells with scaling modifier

### PR DESCRIPTION
**Pull request by MNG-cataclysm**
#### Summary
Mods "[Magiclysm] Tune enhancement spells with a scaling modifier."

#### Purpose of change
Continue improving spellcasting proficiencies and their effects by giving enhancement spells a scaling modifier, like everything else.

#### Describe the solution
Add a scaling modifier to all enhancement spells and add this to the jmath.

#### Describe alternatives you've considered
Not doing this.

#### Testing
Everything works great, buffs get applied properly.

#### Additional context
Conveyance spells are the only ones remaining at this time to finish, and shouldn't take long.

### CODE REVIEW

File: data/mods/Magiclysm/Spells/animist.json [hunk 1]
```diff
@@ -977,7 +977,7 @@
     "id": "animist_spirit_walking",
     "type": "SPELL",
     "name": { "str": "Spirit-Walking" },
-    "description": "Transform your body to pure spirit, making you invisible and intangible.  You can move incredibly quickly while the spell is in effect and can pass by enemies unseen, but you will be unable to affect the physical world or cast spells.",
+    "description": "Tranform your body to pure spirit, making you invisible and intangible.  You can move incredibly quickly while the spell is in effect and can pass by enemies unseen, but you will be unable to affect the physical world or cast spells.",
```
Comment (Hyperseeker) [inline data/mods/Magiclysm/Spells/animist.json:980]: Looks like a misspelling snuck in here.

File: data/mods/Magiclysm/Spells/attunements/Magnetism_Mage.json [hunk 2]
```diff
@@ -261,7 +261,7 @@
     "id": "robot_disabler",
     "type": "SPELL",
     "name": "Surge mod moves",
-    "description": "Subtract some moves from targeted_monster_ids monsters.  Subspell; you aren't allowed to see it except in debug mode.",
+    "description": "Substract some moves from targeted_monster_ids monsters.  Subspell; you aren't allowed to see it except in debug mode.",
```
Comment (Hyperseeker) [inline data/mods/Magiclysm/Spells/attunements/Magnetism_Mage.json:264]: Looks like a misspelling.

**Comment by Hyperseeker** [comment]
Aside from the misspellings, it looks as though this PR is based on an outdated version of the game. For example, it uses `FATIGUE` instead of `SLEEPINESS`.

**Comment by MNG-cataclysm** [comment]
Well, I fixed the spelling issues and the Fatigue -> Sleepiness problem.

**Comment by rty275** [comment]
Weird request, but could you by any chance add some documentation in a comment in the jmath file or somewhere to describe how exactly to apply the scaling proficiency thing for new additions for the future? I could probably figure out via trial and error, but it'd make it easier on me and future contributors if we didn't have to have the skills/know-how to read the math to figure out how it works.

**Pull request merged**